src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -892,6 +900,13 @@ def _maybe_set_compile(self):
                 )
             self.config.reference_compile = False
 
+        if self.device.type == "cpu":
+            logger.warning_once(
```

Comment:
I think we should wrap this in `if self.config.reference_compile:` just like above so users don't get warned when they never manually specified anything related to `reference_compile`.

Unless you'd rather always give this warning so people aren't confused why there's a small precision-esque difference between GPU setups?

Reply:
That was my intention, I forgot the if statement. Fixed in ab11657.

Reply:
Nice! Resolved